gallery/plot_visualization_utils.py

```diff
@@ -0,0 +1,128 @@
+"""
+=======================
+Visualization utilities
+=======================
+
+This example illustrates some of the utilities that torchvision offers for
+visualizing images, bounding boxes, and segmentation masks.
+"""
+
+
+import torch
+import numpy as np
+import scipy.misc
+import matplotlib.pyplot as plt
+
+import torchvision.transforms as T
+
+
+plt.rcParams["savefig.bbox"] = 'tight'
+
+
+def show(img):
+    plt.imshow(np.transpose(img, (1, 2, 0)))
+    plt.axis('off')
+
+
+####################################
+# Visualizing a grid of images
+# ----------------------------
+# The :func:`~torchvision.utils.make_grid` function can be used to create a
+# tensor that represents multiple images in a grid.  This util requires a single
+# image of dtype ``uint8`` as input.
+
+from torchvision.utils import make_grid
+
+racoon = T.ToTensor()(scipy.misc.face().copy())
+racoon_int = T.ConvertImageDtype(dtype=torch.uint8)(racoon)
+grid = make_grid([racoon_int, racoon_int, racoon_int])
+show(grid)
+
+####################################
+# Visualizing Bounding Boxes
+# --------------------------
+# We can use :func:`~torchvision.utils.draw_bounding_boxes` to draw boxes on an
+# image. We can set the colors, labels, width as well as font and font size !
+# The boxes are in ``(xmin, ymin, xmax, ymax)`` format
+# from torchvision.utils import draw_bounding_boxes
+
+from torchvision.utils import draw_bounding_boxes
+
+
+boxes = torch.tensor([[100, 400, 500, 740], [500, 200, 800, 580]], dtype=torch.float)
+labels = ["grass", "racoon"]
+colors = ["blue", "yellow"]
+result = draw_bounding_boxes(racoon_int, boxes, labels=labels, colors=colors, width=10)
+show(result)
+
+#####################################
+# Naturally, we can also plot bounding boxes produced by torchvision detection
+# models.  Here is demo with a Faster R-CNN model loaded from
+# :func:`~torchvision.models.detection.fasterrcnn_resnet50_fpn`
+# model. You can also try using a RetinaNet with
+# :func:`~torchvision.models.detection.retinanet_resnet50_fpn`.
+
+from torchvision.models.detection import fasterrcnn_resnet50_fpn
+
+
+model = fasterrcnn_resnet50_fpn(pretrained=True, progress=False)
+model = model.eval()
+
+outputs = model(racoon.unsqueeze(0))
```

Comment:
Can we use an image which contains categories present in COCO, like dog or people? The current predictions seems a bit arbitrary.

Also, I think it would be good to showcase how to add `labels` (containing for example the class name and score).